code/modules/power/singularity/narsie.dm

```diff
@@ -62,16 +59,15 @@
 	godsmack(A)
 	return
 
-/obj/singularity/narsie/proc/godsmack(var/atom/A)
-	if(istype(A,/obj/))
+/obj/singularity/narsie/proc/godsmack(var/atom/A) //you dare stand before a goddess?!
+	if(isobj(A))
 		var/obj/O = A
 		O.ex_act(1.0)
-		if(O) qdel(O)
-
-	else if(isturf(A))
-		var/turf/T = A
-		T.ChangeTurf(/turf/simulated/floor/plasteel/cult)
-
+		if(O && !qdeleted(O))
```

Comment:
while this is a nifty proc, it's pretty inconsistent with the rest of the codebase; everywhere else explicitly checks for gc_destroyed.